cloudinit/sources/__init__.py

```diff
@@ -1154,8 +1154,8 @@ def parse_cmdline():
     Passing by command line overrides runtime datasource detection
     """
     cmdline = util.get_cmdline()
-    ds_parse_1 = re.search(r"ci\.ds=([a-zA-Z]+)(\s|$)", cmdline)
-    ds_parse_2 = re.search(r"ci\.datasource=([a-zA-Z]+)(\s|$)", cmdline)
+    ds_parse_1 = re.search(r"ci\.ds=([a-zA-Z]+)(\s|$|;)", cmdline)
```

Comment:
+1  on supporting that ';` delimiter more generally, it'll give an option to start shoving in datasource specific config options on the cmdline in the future like metadata_urls,url_retries,url_timeout etc that seem to be fairly well used in multiple datasources.